haystack/core/component/component.py

```diff
@@ -442,7 +447,7 @@ def run(self, value: int):
             instance, {name: OutputSocket(name=name, type=type_) for name, type_ in types.items()}, OutputSocket
         )
 
-    def output_types(self, **types):
+    def output_types(self, **types: Any) -> Callable[[Callable[P, R]], Callable[P, R]]:
```

Comment:
to be able to check the adherence of `run` method to the Protocol, we need to add type hints to the decorator and the decorator factory.